public/app/features/correlations/utils.ts

```diff
@@ -90,3 +91,19 @@ export const createCorrelation = async (
 ): Promise<CreateCorrelationResponse> => {
   return getBackendSrv().post<CreateCorrelationResponse>(`/api/datasources/uid/${sourceUID}/correlations`, correlation);
 };
+
+const getDSInstanceForPane = async (pane: ExploreItemState) => {
+  if (pane.datasourceInstance?.meta.mixed) {
+    return await getDataSourceSrv().get(pane.queries[0].datasource);
+  } else {
+    return pane.datasourceInstance;
+  }
+};
+
+export const generateDefaultLabel = async (sourcePane: ExploreItemState, targetPane: ExploreItemState) => {
```

Comment:
Ideally correlations utils should not rely on any Explore specific types. This function could get `datasourceRef` passed instead of entire `ExploreItemState`.